app/controllers/results_controller.rb

```diff
@@ -156,7 +156,27 @@ def rankings
 
     else
       flash[:danger] = t(".unknown_show")
-      redirect_to rankings_path
+      return redirect_to rankings_path
+    end
+
+    @ranking_timestamp = ComputeAuxiliaryData.successful_start_date || Date.current
+
+    respond_to do |format|
+      format.html {}
+      format.json do
+        cached_data = Rails.cache.fetch ["results-page-api", *@cache_params, @ranking_timestamp] do
+          rows = DbHelper.execute_cached_query(@cache_params, @ranking_timestamp, @query)
+          comp_ids = rows.map { |r| r["competitionId"] }.uniq
+          if @is_by_region
+            rows = compute_rankings_by_region(rows, @continent, @country)
+          end
+          competitions_by_id = Competition.where(id: comp_ids).index_by(&:id).transform_values { |comp| comp.as_json(methods: %w[country], include: [], only: %w[cellName id]) }
```

Comment:
Do you really need the whole country? Usually the `iso2` is enough, pulling the `country.name` from the backend to the frontend results in wrong results because of the stateful locale on API servers.

Reply:
You can then use the `iso2` in the frontend to access even the name via `countries.byIso2[some_iso2].name` or such

Reply:
well the issue is that you can only get the countryId and not the iso2 without calling the country method. I introduced a countries.byId to fix this, which I need to use in multiple places anyway